crates/services/p2p/src/p2p_service.rs

```diff
@@ -219,19 +212,22 @@ impl<Codec: NetworkCodec> FuelP2PService<Codec> {
         let peer_id = match peer_id {
             Some(peer_id) => peer_id,
             _ => {
-                let connected_peers = self.get_peers_info();
-                if connected_peers.is_empty() {
+                let mut peers = self.get_peers_ids();
+                let peers_count = self.swarm.behaviour().total_peers_connected();
+
+                if peers_count == 0 {
                     return Err(RequestError::NoPeersConnected)
                 }
-                let rand_index = rand::thread_rng().gen_range(0..connected_peers.len());
-                *connected_peers.keys().nth(rand_index).unwrap()
+
+                let rand_index = rand::thread_rng().gen_range(0..peers_count);
```

Comment:
consider using [SliceRandom](https://docs.rs/rand/latest/rand/seq/trait.SliceRandom.html)

Reply:
I think maybe [IteratorRandom](https://docs.rs/rand/latest/rand/seq/trait.IteratorRandom.html) would make more sense as the index is just used in an iterator below.